bot/cogs/bot.py

```diff
@@ -60,6 +62,20 @@ def __init__(self, bot: AutoShardedBot):
 
         await ctx.invoke(self.info)
 
+    async def on_message(self, msg: Message):
+        if msg.content.count("\n") >= 3:  # more than three lines
+            try:
+                tree = ast.parse(msg.content)
+
+                # Attempts to parse the message into an AST node.
+                # Invalid Python code will raise a SyntaxError.
+                # => text which is not Python code will be filtered.
+                if not all(isinstance(node, ast.Expr) for node in tree.body):  # we dont want hi\nthere\nguys\nD
+                    await self.bot.send_message(msg.author, "```python\n{}\n```".format(msg.content))
```

Comment:
send_message() is pre-rewrite, and I'm pretty sure you want to send it to the channel, so the rewrite version would be msg.channel.send(<message here>)
Also if the message (with the unformatted python code) were >1986 characters this line would throw an error, maybe add a check against this?
(adding again because previous comment is marked as out of date, when the line hasn't been fixed)